src/ui/CategoryFacet/CategoryFacetSearch.ts

```diff
@@ -219,4 +223,14 @@ export class CategoryFacetSearch {
         .replace(regex, '<span class="coveo-highlight">$1</span>');
     });
   }
+  private logAnalyticsEvent() {
```

Comment:
I'm thinking if it would be better to centralize these calls to analytics in the root `CategoryFacet` component. The implementation inside the `CategoryFacet` might look something like:

```
public logAnalyticsEvent(eventName: string, dataObj: Record<string> = {}) {
    const data = {
       categoryFacetId: this.options.id,
       categoryFacetTitle: this.options.title,
       ...dataObj
   }
    this.usageAnalytics.logCustomEvent(name, data);
}
```

and then the usage in CategoryFacetSearch, CategoryValue etc. would be:
e.g.

```
    this.categoryFacet.logAnalyticsEvent(analyticsActionCauseList.categoryFacetSearch)
```

Reply:
+1